File: Linguagem C/idades.c
#include <stdio.h>
#include <locale.h>

void limparEntrada() {
    char c;
    while((c = getchar()) != '\n' && c != EOF) {}
}

void ler_texto(char *buffer, int length) {
    fgets(buffer, length, stdin);
    strtok(buffer, "\n");
}

int main() {

    setlocale(LC_ALL, "Portuguese");
    char nome[50], nome2[50];
    int idade, idade2;
    double mediaIdade;

    printf("Dados da primeira pessoa\n");

    printf("Nome: ");
    ler_texto(nome, 50);

    printf("Idade: ");
    scanf("%d", &idade);

    printf("\nDados da segunda pessoa:\n");

    printf("Nome: ");
    limparEntrada();
    ler_texto(nome2, 50);

    printf("Idade: ");
    scanf("%d", &idade2);

    mediaIdade = (double)(idade + idade2) / 2;
    printf("A idade media de %s e %s é de %.1lf", nome, nome2, mediaIdade);

return 0;
}

File: Java/idades.java
import java.util.Locale;
import java.util.Scanner;

public class idades {
    public static void main(String[] args) {

        Locale.setDefault(Locale.US);
        Scanner input = new Scanner(System.in);

        String nome1, nome2;
        int idade1, idade2;

        System.out.println("Dados da primeira pessoa:");
        System.out.print("Nome: ");
        nome1 = input.nextLine();
        System.out.print("Idade: ");
        idade1 = input.nextInt();

        System.out.println("Dados da segunda pessoa:");
        System.out.print("Nome: ");
        nome2 = input.nextLine();
        System.out.print("Idade: ");
        idade2 = input.nextInt();

        float media;
        media = (idade1 + idade2) / 2;

        System.out.print("A idade média de " + nome1 + " e " + nome2 + " é de " + String.format("%.1f", media) + " anos");

    }
}
